domain/lib/repositories/auth_repository.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:domain/domain.dart';
+
+abstract class AuthRepository {
+  Future<AppUser> signIn({
+    required String email,
+    required String password,
+  });
+
+  Future<AppUser> signInUsingGoogleAcc();
+
+  Future<AppUser> signUp({
+    required String name,
+    required String email,
+    required String password,
+  });
+
+  Future<void> signOut();
+
+  Future<AppUser> getStoredUser();
```

Comment:
i think u should separate this interface because its break SRP principle. U should create something like this:

![изображение](https://github.com/Burvan/Flutter_lab_project/assets/70040363/91cdfd71-ddf1-458c-8677-41f932231e8d)